### PR TITLE
Simulation functionality TLM systems

### PR DESCRIPTION
**Pull request by robbr48**
### Purpose

Implement simulation functions for TLM systems.

### Approach

Implemented functions:
- `SystemTLM::instantiate`
- `SystemTLM::initialize`
- `SystemTLM::simulate`
- `SystemTLM::terminate`
- `SystemTLM::initializeSubSystem`
- `SystemTLM::simulateSubSystem`

These changes are untested, but they compile and do not break existing tests.

### Type of Change

- New feature (non-breaking change which adds functionality)

### Checklist

- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by robbr48** [comment]
@lochel Do you want to take a look at this?

File: src/OMSimulatorLib/SystemTLM.h [hunk 1]
```diff
@@ -55,6 +55,7 @@ namespace oms3
 
     oms_status_enu_t instantiate();
     oms_status_enu_t initialize();
+    oms_status_enu_t simulate();
```
[the remark this answers is not in the document]
Comment (lochel) [reply src/OMSimulatorLib/SystemTLM.h]: Note, `simulate()` is actually the same as `stepUntil(getModel()->getStopTime())`.

[the remark this answers is not in the document]
Comment (robbr48) [reply src/OMSimulatorLib/SystemTLM.h]: But doesn't "stepUntil" give the impression than you can keep simulating from where you stopped? I.e. call stepUntil multiple times in a row. This is not possible for TLM simulations. You can only run a complete simulation.

File: src/OMSimulatorLib/SystemTLM.cpp [hunk 2]
```diff
@@ -386,6 +451,21 @@ oms_status_enu_t oms3::SystemTLM::updateInitialValues(const oms3::ComRef cref)
...
+  oms_status_enu_t status = getSubSystem(cref)->initialize();
+  if(status == oms_status_ok) {
+    initializedsubsystems.push_back(cref);
+  }
+  return status;
+}
+
+oms_status_enu_t oms3::SystemTLM::simulateSubSystem(oms3::ComRef cref)
+{
+  oms_status_enu_t status = getSubSystem(cref)->stepUntil(parentModel->getStopTime());
```
[the remark this answers is not in the document]
Comment (robbr48) [reply src/OMSimulatorLib/SystemTLM.cpp]: Ok, I fixed it.

File: src/OMSimulatorLib/SystemTLM.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (robbr48) [reply src/OMSimulatorLib/SystemTLM.h]: Ok, I changed it anyway. It works as long as we don't have a oms3_stepUntil function in the API. Consider the following pseudo script:
`oms3_setStartTime(0)`
`oms3_setStopTime(10)`
`oms3_stepUntil(5)` <- simulates from 0 to 5
`oms3_stepUntil(10)` <- simulates from 0 to 10 (i.e. restarts simulation)
However, the user probably expects it to first simulate from 0 to 5, and then from 5 to 10. Which is not possible for TLM models.

**Comment by OpenModelica-Hudson** [comment]
The following pull requests are now being [tested](https://test.openmodelica.org/hudson/job/OMSimulator-PR/750//console). It may take a while, depending on the workload.
  - OpenModelica/OMSimulator#441

The changes will be pushed if all tests run successfully.

File: src/OMSimulatorLib/System.h [hunk 3]
```diff
@@ -132,6 +132,10 @@ namespace oms3
 
     DirectedGraph initialUnknownsGraph;
     DirectedGraph outputsGraph;
+
+  protected:
+
+
```
Comment (lochel) [inline src/OMSimulatorLib/System.h]: Don't push empty changes...

**Comment by OpenModelica-Hudson** [comment]
The following pull requests are now being [tested](https://test.openmodelica.org/hudson/job/OMSimulator-PR/752//console). It may take a while, depending on the workload.
  - OpenModelica/OMSimulator#441

The changes will be pushed if all tests run successfully.

**Comment by OpenModelica-Hudson** [comment]
**[Windows-GCC:](https://test.openmodelica.org/hudson//job/OMSimulator-PR-Windows/524)** The test suite is unstable. :-1:

**Comment by OpenModelica-Hudson** [comment]
**[Linux:](https://test.openmodelica.org/hudson/job/OMSimulator-PR/757/)** The test suite is unstable. :worried:

**Comment by OpenModelica-Hudson** [comment]
The following pull requests are now being [tested](https://test.openmodelica.org/hudson/job/OMSimulator-PR/759//console). It may take a while, depending on the workload.
  - OpenModelica/OMSimulator#441

The changes will be pushed if all tests run successfully.

**Comment by OpenModelica-Hudson** [comment]
All tests were successful. :smile:

**Pull request closed without merging**